src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -390,6 +390,135 @@ BOOST_AUTO_TEST_CASE(SimpleLinearLayerTest)
   BOOST_REQUIRE_EQUAL(arma::accu(delta), 0);
 }
 
+/**
+ * Linear module weight initialization test.
+ */
+BOOST_AUTO_TEST_CASE(LinearLayerWeightInitializationTest)
+{
+  size_t inSize = 10, outSize = 4;
+  Linear<>* linear = new Linear<>(inSize, outSize);
+  linear->Reset();
+  RandomInitialization().Initialize(linear->Weight(), outSize, inSize);
+  linear->Bias().ones();
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(linear->Weight().begin(),
+      linear->Weight().end(), linear->Parameters().begin(),
+      linear->Parameters().end() - outSize);
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(linear->Bias().begin(),
+      linear->Bias().end(), linear->Parameters().begin() + inSize * outSize,
+      linear->Parameters().end());
+
+  BOOST_REQUIRE_EQUAL(linear->Weight().n_rows, outSize);
+  BOOST_REQUIRE_EQUAL(linear->Weight().n_cols, inSize);
+  BOOST_REQUIRE_EQUAL(linear->Bias().n_rows, outSize);
+  BOOST_REQUIRE_EQUAL(linear->Bias().n_cols, 1);
+  BOOST_REQUIRE_EQUAL(linear->Parameters().n_rows, inSize * outSize + outSize);
+
+  delete linear;
+}
+
+/**
+ * Atrous Convolution module weight initialization test.
+ */
+BOOST_AUTO_TEST_CASE(AtrousConvolutionLayerWeightInitializationTest)
+{
+  size_t inSize = 2, outSize = 3;
+  size_t kernelWidth = 4, kernelHeight = 5;
+  AtrousConvolution<>* module = new AtrousConvolution<>(inSize, outSize,
+      kernelWidth, kernelHeight, 6, 7, std::make_tuple(8, 9),
+      std::make_tuple(10, 11), 12, 13, 14, 15);
+  module->Reset();
+  RandomInitialization().Initialize(module->Weight(), kernelWidth, kernelHeight,
+      inSize * outSize);
```

Comment:
```suggestion
  module->Weight().randu();
```

Reply:
Perhaps someone wants to apply some other init rules like `GlorotInitialization` on weight and keep `bias` to zero. How can we accomplish it? Other init rules also have almost the same implementation.

Reply:
> Perhaps someone wants to apply some other init rules like GlorotInitialization on weight and keep bias to zero.

I didn't get the point,  `module->Weight().randu();` doesn't affect the bias.

Reply:
> Other init rules also have almost the same implementation.

In that case we need a separate PR for the changes.

Reply:
>I didn't get the point, module->Weight().randu(); doesn't affect the bias.

I mean we do have `x.randu()` from armadillo but how will use other init rules. The other thing which is currently used is initializing both `weight` and `bias` with the same initialization rules. Like we do in `FFN<LossType, InitType>`, it initializes both `weight` and `bias` with the same rule.

If we were to just do `module->Weight().randu()` and `module->Bias().randu()`, there was no need for separate methods,
we can do `module->Parameters().randu()`.

From https://cs231n.github.io/neural-networks-2/
>Initializing the biases. It is possible and common to initialize the biases to be zero since the asymmetry breaking is provided by the small random numbers in the weights. For ReLU non-linearities, some people like to use small constant value such as 0.01 for all biases because this ensures that all ReLU units fire in the beginning and therefore obtain and propagate some gradient. However, it is not clear if this provides a consistent improvement (in fact some results seem to indicate that this performs worse) and it is more common to simply use 0 bias initialization.

Reply:
I get your point, thanks for the clarification.